application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -0,0 +1,224 @@
+import org.togetherjava.tjbot.commands.utils.MessageUtils;
+import org.togetherjava.tjbot.config.Config;
+
+import java.awt.*;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+
+/**
+ * Implements the /report command, which allows users to report a selected offensive message from
+ * another user. The message is then forwarded to moderators in a dedicated channel given by
+ * {@link Config#getModMailChannelPattern()}.
+ */
+public final class ReportCommand extends BotCommandAdapter implements MessageContextCommand {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReportCommand.class);
+    private static final String COMMAND_NAME = "report";
+    private static final String REPORT_REASON_INPUT_ID = "reportReason";
+    private static final int COOLDOWN_DURATION_VALUE = 3;
+    private static final ChronoUnit COOLDOWN_DURATION_UNIT = ChronoUnit.MINUTES;
+    private static final Color AMBIENT_COLOR = Color.BLACK;
+    private final Cache<Long, Instant> authorToLastReportInvocation = createCooldownCache();
+    private final Predicate<String> modMailChannelNamePredicate;
+    private final String configModMailChannelPattern;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param config to get the channel to forward reports to
+     */
+    public ReportCommand(Config config) {
+        super(Commands.message(COMMAND_NAME), CommandVisibility.GUILD);
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+
+        configModMailChannelPattern = config.getModMailChannelPattern();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        long userID = event.getUser().getIdLong();
+        String reportedMessageTimestamp = event.getTarget().getTimeCreated().toInstant().toString();
+
+        if (handleIsOnCooldown(event)) {
+            return;
+        }
+        authorToLastReportInvocation.put(userID, Instant.now());
+
+        String reportedMessage = event.getTarget().getContentRaw();
+        String reportedMessageID = event.getTarget().getId();
+        String reportedMessageChannel = event.getTarget().getChannel().getId();
+        String reportedAuthorName = event.getTarget().getAuthor().getName();
+        String reportedAuthorAvatarURL = event.getTarget().getAuthor().getAvatarUrl();
+        String reportedAuthorID = event.getTarget().getAuthor().getId();
+
+        TextInput body = TextInput
```

Comment:
y naming it `body`?? name it `reasonIntput` or similar

Reply:
`modalBody` ?

Reply:
no

Reply:
But its not an input, it builds the modal text

Reply:
it is an input

Reply:
Yeah your right, just smth kind of feels of and I can't "put my finger on it"

Reply:
Because it doesn't just handles the text field, it handles bit above it